src/utils/src/Thread.c

```diff
@@ -150,16 +177,71 @@ PUBLIC_API TID defaultGetThreadId()
     return (TID) pthread_self();
 }
 
+PUBLIC_API STATUS defaultCreateThreadWithParams(PTID pThreadId, startRoutine start, SIZE_T stackSize, PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    pthread_t threadId;
+    INT32 result;
+    pthread_attr_t* pAttr = NULL;
+
+    CHK(pThreadId != NULL, STATUS_NULL_ARG);
+
+    pthread_attr_t attr;
+    pAttr = &attr;
+    result = pthread_attr_init(pAttr);
+    CHK_ERR(result == 0, STATUS_THREAD_ATTR_INIT_FAILED, "pthread_attr_init failed with %d", result);
+    result = pthread_attr_setstacksize(&attr, stackSize);
+    CHK_ERR(result == 0, STATUS_THREAD_ATTR_SET_STACK_SIZE_FAILED, "pthread_attr_setstacksize failed with %d", result);
+
+    result = pthread_create(&threadId, pAttr, start, args);
+    switch (result) {
+        case 0:
+            // Successful case
+            break;
+        case EAGAIN:
+            CHK(FALSE, STATUS_THREAD_NOT_ENOUGH_RESOURCES);
+        case EINVAL:
+            CHK(FALSE, STATUS_THREAD_INVALID_ARG);
+        case EPERM:
+            CHK(FALSE, STATUS_THREAD_PERMISSIONS);
+        default:
+            // Generic error
+            CHK(FALSE, STATUS_CREATE_THREAD_FAILED);
+    }
+
+    *pThreadId = (TID) threadId;
+
+CleanUp:
+
+    if (pAttr != NULL) {
+        result = pthread_attr_destroy(pAttr);
+        if (result != 0) {
+            DLOGW("pthread_attr_destroy failed with %u", result);
+        }
+    }
+
+    return retStatus;
+}
+
 PUBLIC_API STATUS defaultCreateThread(PTID pThreadId, startRoutine start, PVOID args)
```

Comment:
Would be ideal if this can call `defaultCreateThreadWithParams` and just specify the thread size it needs, so we don't have the rest of the logic in two places.

Reply:
done